### PR TITLE
Adding an ISODateTime and ID property to all data objects.

### PR DESCRIPTION
**Pull request by cflynn3**

### CODE REVIEW

**Comment by cflynn3** [comment]
A few changes here, all related to the `DataObjectMixin` base class.

- All data objects now have an ID property that is a UUID automatically generated on instantiation of the derived class.
- All data objects now have an `ISODateTime` property that displays the date in standard ISO format. It is updated automatically every time the date is changed.
- Fixed various existing bugs related to ID and dates.

**Review by Dan-work1 — APPROVED** [review]

File: ResSimpy/Completion.py [hunk 1]
```diff
@@ -103,10 +99,11 @@ def __init__(self, date: str, i: Optional[int] = None, j: Optional[int] = None,
...
             unit_system: Optional[UnitSystem]: The unit system to use for the completion.
         """
-        super().__init__({})
-        self._date_format = date_format
+        super().__init__()
         self.__well_radius = well_radius
-        self.__date = date
+        self._date_format = date_format
+        self._start_date = start_date
+        self.date = date
```
Comment (jonnyh404) [inline ResSimpy/Completion.py:106]: Can these be passed through to the init of the super class?

File: ResSimpy/Nexus/DataModels/Network/NexusConstraint.py [hunk 2]
```diff
@@ -194,15 +195,43 @@ class NexusConstraint(Constraint):
...
+                self.__setattr__(f"_{attribute}", properties_dict[attribute])
+
+        if date_format is not None:
+            self._date_format = date_format
+
+        if start_date is not None:
+            self._start_date = start_date
+
+        if unit_system is not None:
+            self._unit_system = unit_system
```
Comment (jonnyh404) [inline ResSimpy/Nexus/DataModels/Network/NexusConstraint.py:222]: Can we put these into the super class' init instead?

File: ResSimpy/Nexus/DataModels/Network/NexusNode.py [hunk 3]
```diff
@@ -15,16 +17,43 @@ class NexusNode(Node):
...
+                self.__setattr__(f"_{attribute}", properties_dict[attribute])
+
+        if date_format is not None:
+            self._date_format = date_format
+
+        if start_date is not None:
+            self._start_date = start_date
+
+        if unit_system is not None:
+            self._unit_system = unit_system
```
Comment (jonnyh404) [inline ResSimpy/Nexus/DataModels/Network/NexusNode.py:44]: As above.

**Review by jonnyh404 — APPROVED** [review]
Looks good to me. Think about handling the setting on the parent classes rather than at the children class level.

**Pull request merged**